src/include/index/ivf_pq_index.h

```diff
@@ -291,15 +293,17 @@ class ivf_pq_index {
       size_t max_iter = 2,
       float tol = 0.000025,
       std::optional<TemporalPolicy> temporal_policy = std::nullopt,
-      uint64_t seed = std::random_device{}())
+      uint64_t seed = std::random_device{}(),
+      DistanceMetric distance_metric = DistanceMetric::L2)
```

Comment:
(Minor) Probably best to put `distance_metric` before `seed` because it's much more common to set `distance_metric` than it is to set `seed`.

Reply:
That would make us fail the backwards compatibility test unfortunately - Unless we want to change the test it has to come after seed (I think)

Reply:
Mmm that doesn't really make sense to me, can you explain more?